### ISSUE TITLE
Validating the model

### ISSUE THREAD

**Issue by SURABHI-GUPTA** [AliaksandrSiarohin/first-order-model#429]
@AliaksandrSiarohin It seems, validation is not done along with the training pipeline. Is there any specific reason for this?
What changes do we need to do for adding that to the training pipeline?

**Comment by AliaksandrSiarohin** [comment]
It will take significant amount of time, for example generation all videos for vox test set is 8 hours. Also training loss has pretty good correlation with final accuracy, so intermediate validation is not really useful.

**Comment by SURABHI-GUPTA** [comment]
@AliaksandrSiarohin 
Okay. I tried adding validation code. 
I did like:
after training loop, I added
generator.eval()        ## only for generator
kp_regressor_full = KPRegressorFull(kp_detector, generator, discriminator, train_params)
if torch.cuda.is_available():
   kp_regressor_full = DataParallelWithCallback(kp_regressor_full, device_ids=device_ids)

and then again for training
generator.train()
kp_regressor_full = KPRegressorFull(kp_detector, generator, discriminator, train_params)
if torch.cuda.is_available():
   kp_regressor_full = DataParallelWithCallback(kp_regressor_full, device_ids=device_ids)

Is it okay and do we need to set,
with torch.no_grad():

**Comment by AliaksandrSiarohin** [comment]
Sorry, but your code makes no sense to me. You can found how validation is implemented in reconstruction.py
